### ISSUE TITLE
Issue with automatic start

### ISSUE THREAD

**Issue by Sh00by** [carljm/synology-mediamon#11]
Hi,

The scripts works properly by manually using it. But I'm having an issue with automatic start (using /usr/local/etc/rc.d/ or the syno Scheduled Task) and I'm not enough skilled in Python to understand what's wrong....

```
File "/volume1/homes/admin/mediamon.py", line 134, in <module>
    exclude_filter=lambda p: '/@' in p
  File "/volume1/@appstore/py3k/usr/local/lib/python3.5/site-packages/pyinotify.py", line 1916, in add_watch
    exclude_filter)
  File "/volume1/@appstore/py3k/usr/local/lib/python3.5/site-packages/pyinotify.py", line 1833, in __add_watch
    wd = self._inotify_wrapper.inotify_add_watch(self._fd, path, mask)
  File "/volume1/@appstore/py3k/usr/local/lib/python3.5/site-packages/pyinotify.py", line 153, in inotify_add_watch
    return self._inotify_add_watch(fd, pathname, mask)
  File "/volume1/@appstore/py3k/usr/local/lib/python3.5/site-packages/pyinotify.py", line 246, in _inotify_add_watch
    pathname = pathname.encode(sys.getfilesystemencoding())
UnicodeEncodeError: 'ascii' codec can't encode characters in position 42-43: ordinal not in range(128)
```

By the way, on my Syno I have both python 2.7 (for another app)  and python 3.
Model DS214play / FW : DSM 6.0.2-8451 Update 2

**Comment by carljm** [comment]
Where do you see this error traceback?

**Comment by Sh00by** [comment]
I activated the mail notification in the syno task scheduler (for launching the script) when starting or shutdowning the syno. This is the only way I found to get logs

**Comment by Sh00by** [comment]
Ok, I found the issue ... I don't know why really.....Probably because I have files/ folder with accent..
Found it from the original post https://codesourcery.wordpress.com/2012/11/29/more-on-the-synology-nas-automatically-indexing-new-files/

Vincent Barberet
April 28, 2016 at 9:44 pm
Well, I’ve found a dirty workaround.
The issue is that I have some filenames that are not UTF-8 encoded, but my filesystem is UTF-8 (use python command “sys.getfilesystemencoding()” to get your filesystem encoding).
So, the real issue is that I have bad filenames. As I cannot fix easily those filenames, I have to live with them.

Here is what I did:
in /volume1/@appstore/py3k/usr/local/lib/python3.5/site-packages/pyinotify.py, line 246:
I have put:
pathname = pathname.encode(sys.getfilesystemencoding(), ‘surrogateescape’)
instead of
pathname = pathname.encode(sys.getfilesystemencoding())
